src/WebJobs.Script/Description/DotNet/FunctionAssemblyLoader.cs

```diff
@@ -63,14 +63,30 @@ internal Assembly ResolveAssembly(object sender, ResolveEventArgs args)
                 // This allows us to correctly handle retargetable assemblies, redirects, etc.
                 if (result == null)
                 {
-                    string assemblyName = ((AppDomain)sender).ApplyPolicy(args.Name);
+                    var appDomain = (AppDomain)sender;
+                    string assemblyName = appDomain.ApplyPolicy(args.Name);
 
                     // If after applying the current policy, we now have a different target assembly name, attempt to load that 
                     // assembly
                     if (string.Compare(assemblyName, args.Name) != 0)
                     {
                         result = Assembly.Load(assemblyName);
                     }
+
+
+                    if (context == null && result == null)
+                    {
+                        // Fallback. 
+                        var shortName = new AssemblyName(assemblyName).Name;
```

Comment:
@MikeStall I'd be happy to pull your branch down to debug the issues you ran into so we can come up with a safe solution.